proto/tests/server/test_pi_server.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017, Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <PI/proto/pi_server.h>
+#include <string>
+
+namespace pi {
+namespace proto {
+namespace testing {
+namespace {
+
+static std::string grpc_server_addr = "0.0.0.0";
+static std::string grpc_server_port = "50051";
```

Comment:
please use `50054`. It's a little bit hacky at the moment but each test file has to use a different port number in case the tests are run in parallel. `50051...50053` are already taken by the other files.
long term, we definitely need a better way of allocating ports for testing. Either something like this (https://github.com/grpc/grpc/blob/master/test/core/util/port.h) or something simpler by creating unique temporary directories.

Reply:
Fixed. Thanks for the pointer to port.h